src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -0,0 +1,133 @@
+import {
+    DatasetNavigationInterface,
+    DatasetViewTypeEnum,
+} from "./../../dataset-view/dataset-view.interface";
+import { ActivatedRoute, ParamMap } from "@angular/router";
+import { ModalService } from "./../../components/modal/modal.service";
+import { BaseComponent } from "src/app/common/base.component";
+import { NavigationService } from "src/app/services/navigation.service";
+import {
+    ChangeDetectionStrategy,
+    Component,
+    OnInit,
+    ViewEncapsulation,
+} from "@angular/core";
+import { promiseWithCatch, requireValue } from "src/app/common/app.helpers";
+import { searchAdditionalButtonsEnum } from "src/app/search/search.interface";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import ProjectLinks from "src/app/project-links";
+
+@Component({
+    selector: "app-metadata-block",
+    templateUrl: "./metadata-block.component.html",
+    styleUrls: ["./metadata-block.component.sass"],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetadataBlockComponent extends BaseComponent implements OnInit {
+    public datasetInfo: DatasetInfo;
+    public datasetViewType = DatasetViewTypeEnum.History;
+
+    constructor(
+        private navigationService: NavigationService,
+        private modalService: ModalService,
+        private activatedRoute: ActivatedRoute,
+    ) {
+        super();
+    }
+    ngOnInit(): void {
+        this.datasetInfo = this.getDatasetInfoFromUrl();
+    }
+
+    public showOwnerPage(): void {
```

Comment:
A few functions here are duplicating code from `DatasetComponent`. Introduce an intermediate base class to share common code between `DatasetComponent` and `MetadataBlockComponent`. Note that it cannot be placed to `BaseComponent`, which is a higher level generalization. It has to be something more specific.

Reply:
Added new class to share common code between DatasetComponent and MetadataBlockComponent